pkg/kapp/diff/change_set_with_versioned_rs.go

```diff
@@ -225,9 +234,10 @@ func (d ChangeSetWithVersionedRs) newNoopChange(existingRes ctlres.Resource) Cha
 	return NewChangePrecalculated(existingRes, nil, nil, ChangeOpNoop, nil, OpsDiff{})
 }
 
-func (ChangeSetWithVersionedRs) numOfResourcesToKeep(res ctlres.Resource) (int, error) {
-	// TODO get rid of arbitrary cut off
-	numToKeep := 5
+func (d ChangeSetWithVersionedRs) numOfResourcesToKeep(change Change) (int, error) {
+	res := change.NewOrExistingResource()
```

Comment:
I am not a 100% sure about the side-effects of this, but this should only play a role for "actually" versioned resources, as only those should get `keep` Changes that have no "new" resource.

That actually might need further investigation as the `numOfResourcesToKeep` will now work on `existingRes` for `kee` changes, not `newRes` anymore ... :thinking: